### PR TITLE
Create solution that contains all the projects and build it

### PR DESCRIPTION
**Pull request by zvirja**
Before we built AutoFixture project multiple times as a part of the different solutions. It significantly slowed downs the build process (e.g. because code analysis was run for the AutoFixture project for a few times).

I've created a solution `All.sln` that contains all the projects we have. The benefits are following:
- it takes about half of time to build the project (e.g. 4 min 19 secs compared to average 7 min 30 secs)[https://ci.appveyor.com/project/AutoFixture/autofixture/history]
- it allows to reduce the AppVeyor [build log](https://ci.appveyor.com/project/AutoFixture/autofixture/build/1.0.95) size, so browser lags significantly less when a build is opened (3143 lines compared to 6776 lines).
- you have ability to load all the projects simultaneously and perform some code analysis on the whole codebase (e.g. check usages of the particular API).

~~The only thing I'm not confident in is the solution name. Do you like `AutoFixture.AllProjects` name or the `AutoFixture.All` name looks nicer?~~
Solution name was updated to `All.sln`.

### CODE REVIEW

File: Src/AutoFakeItEasy2.UnitTest/AutoFakeItEasy2.UnitTest.csproj [hunk 1]
```diff
@@ -62,7 +62,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\AutoFakeItEasy2\AutoFakeItEasy2.csproj">
-      <Project>{b6b79e32-ee90-4ae1-bd32-f50b5a0c2b40}</Project>
+      <Project>{68A3F347-7D9D-4020-A3D5-F127234DD429}</Project>
```
Comment (ecampidoglio) [inline Src/AutoFakeItEasy2.UnitTest/AutoFakeItEasy2.UnitTest.csproj]: Why did the project ID have to change?

File: Src/AutoFakeItEasy2/AutoFakeItEasy2.csproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProductVersion>8.0.30703</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{B6B79E32-EE90-4AE1-BD32-F50B5A0C2B40}</ProjectGuid>
+    <ProjectGuid>{68A3F347-7D9D-4020-A3D5-F127234DD429}</ProjectGuid>
```
Comment (ecampidoglio) [inline Src/AutoFakeItEasy2/AutoFakeItEasy2.csproj]: Why did the project ID have to change?

File: Src/AutoFakeItEasyUnitTest/AutoFakeItEasyUnitTest.csproj [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProductVersion>8.0.30703</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{CDEB7E7C-85A0-40F1-80F9-34A9511491F1}</ProjectGuid>
+    <ProjectGuid>{E5CAB3E1-10DD-4081-A5FF-ED8A11F3C2CB}</ProjectGuid>
```
Comment (ecampidoglio) [inline Src/AutoFakeItEasyUnitTest/AutoFakeItEasyUnitTest.csproj]: Why did the project ID have to change?

File: Build.fsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ open System.Text.RegularExpressions
 let releaseFolder = "Release"
 let nunitToolsFolder = "Packages/NUnit.Runners.2.6.2/tools"
 let nuGetOutputFolder = "NuGetPackages"
-let solutionsToBuild = !! "Src/*.sln"
+let solutionsToBuild = !! "Src/AutoFixture.AllProjects.sln"
```
Comment (ecampidoglio) [inline Build.fsx]: I vote for `All.sln` 👍

Comment (zvirja) [inline Build.fsx]: I like this name, so renamed to `All.sln` 😉

File: Src/AutoFakeItEasy2.UnitTest/AutoFakeItEasy2.UnitTest.csproj (hunk 1 above)
Comment (zvirja) [inline Src/AutoFakeItEasy2.UnitTest/AutoFakeItEasy2.UnitTest.csproj]: It appeared that AutoFakeItEasy2 projects were not created correctly - they were copy-pasted from the original AutoFakeItEasy ones.

After I added all projects to the single solution, the conflicts appeared. Therefore, you see 3 modifications - 2 project IDs and single project reference were updated.

File: Src/AutoFakeItEasy2/AutoFakeItEasy2.csproj (hunk 2 above)
Comment (zvirja) [inline Src/AutoFakeItEasy2/AutoFakeItEasy2.csproj]: See [above](https://github.com/AutoFixture/AutoFixture/pull/756#discussion_r113718149)

File: Src/AutoFakeItEasyUnitTest/AutoFakeItEasyUnitTest.csproj (hunk 3 above)
Comment (zvirja) [inline Src/AutoFakeItEasyUnitTest/AutoFakeItEasyUnitTest.csproj]: See [above](https://github.com/AutoFixture/AutoFixture/pull/756#discussion_r113718149)

**Pull request merged**